modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/CurrencySnapshotConsensusFunctions.scala

```diff
@@ -49,7 +52,8 @@ object CurrencySnapshotConsensusFunctions {
       signedArtifact: Signed[CurrencyIncrementalSnapshot],
       context: CurrencySnapshotContext
     ): F[Unit] =
-      stateChannelSnapshotService.consume(signedArtifact, context)
+      stateChannelSnapshotService.consume(signedArtifact, context) >>
+        maybeDataApplication.traverse_(_.onSnapshotAcceptedLive(signedArtifact))
```

Comment:
it should be rather `onConsensusResult` or `consumeSnapshotAfterConsensus` or something like that. it isn't equal to snapshot acceptance.

Reply:
@marcinwadon I can't find reference to either `onConsensusResult` or `consumeSnapshot`. Where are you suggesting this change go?

Reply:
I meant the function name

Reply:
onSnapshotAcceptedLive seems odd to me. It's not triggered on snapshot acceptance, and what does "live" mean in this context?

Reply:
@ryle-goehausen  ☝️

Reply:
I agree that the name seems odd. Could we follow the convention we already have? So maybe: `consumeMajorityArtifact`? Especially that it takes the artifact as input. `onSnapshotAcceptedLive` sounds new and exotic but maybe it should sound plain and keep to the convention rather.

Reply:
Renamed function to `consumeMajorityArtifact`

Reply:
@TheMMaciek @sdavidp i would stick to the convention "onSomething" for all the handlers. We may add more in the future. What do you think?

The same way like you can find on web: onChange, onClick, onDoubleClick, and so on

Reply:
The reason for naming it 'snapshot' is because, from the perspective of the DataApplication (the only place we currently need this,) it's always going to be a currency snapshot. So it's less abstract than an arbitrary artifact since this is a specific implementation. (Since we're making this change in CurrencySnapshotConsensusFns.)

Consume is also okay as nomenclature, I suggested inserting it here because it seemed like an easy place (to avoid any interactions with downloads or overlap in other 'acceptance' style logic,) so maybe that's a better name than acceptedLive? The live was meant to distinguish between the case where we're accepting a snapshot due to a download or rollback or anything else -- other than a live result of a consensus process (this is important in the application context.)

'Consensus' is an appropriate renaming of 'live', `onConsensusResult` is fine here, I do agree with the `on` standard for application handlers (especially when they're so isolated like this.) But you could also insert snapshot in the name since it's not a specific artifact. I.e. `onSnapshotConsensusResult`

Reply:
> The reason for naming it 'snapshot' is because, from the perspective of the DataApplication (the only place we currently need this,) it's always going to be a currency snapshot. So it's less abstract than an arbitrary artifact since this is a specific implementation. (Since we're making this change in CurrencySnapshotConsensusFns.)

No problem, it can be a `snapshot`.

> The live was meant to distinguish between the case where we're accepting a snapshot due to a download or rollback or anything else -- other than a live result of a consensus process (this is important in the application context.)

I'd keep using the unified nomenclature across the codebase. We've never used `Live` for such distinguish and it will be misleading.

@ryle-goehausen I agree with you. I'm fine with both `onConsensusResult` or `onSnapshotConsensusResult`, whatever :) just not `acceptance` since it's not triggered from/during acceptance logic and not `live` as it may be misleading.

Reply:
@sdavidp /\